src/effect.cpp

```diff
@@ -8,10 +8,32 @@
 #include <map>
 #include <sstream>
 
-std::map<std::string, effect_type> effect_types;
+namespace {
+std::map<efftype_id, effect_type> effect_types;
+}
+
+template<>
+const effect_type& string_id<effect_type>::obj() const
+{
+    const auto iter = effect_types.find( *this );
+    if( iter == effect_types.end() ) {
+        debugmsg( "invalid effect type id %s", c_str() );
+        const effect_type dummy{};
```

Comment:
I'm fixing with the merge, but FYI this is undefined behavior (returning a reference to a variable with auto lifetime), I'm making it a static so it stays in scope.